proposed/2022/dotnet-nuget-config-spec.md

```diff
@@ -0,0 +1,79 @@
+
+Currently, there is no NuGet configuration CLI for dotnet.exe. It's inconvenient for NuGet users to know about the NuGet configuration file locations and figure out where is the merged configuration coming from. 
+
+## Who are the customers
+
+This feature is for dotnet.exe users.
+
+## Goals
+Design and implement `dotnet nuget config` command.
+
+## Non-Goals
+Design and implement `dotnet nuget config` command with commands other than `list`, E.g. add/update/delete
+
+## Solution
+The following command will be implemented in the `dotnet.exe` CLI.
+
+### `dotnet nuget config`
+
+#### Commands
+
+- List
+
+Lists all the NuGet configuration file locations. This command will include all the NuGet configuration file that will be applied, when invoking NuGet command from the current working directory path. The listed NuGet configuration files are in priority order. So the order of loading those configurations is reversed, that is, loading order is from the bottom to the top. So the configuration on the top will apply.
+You may refer to [How settings are applied](https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#how-settings-are-applied) for more details. 
+
+#### Arguments
+
+- CURRENT_DIRECTORY
+
+Run this command as if current directory is set to the specified directory.
+
+#### Options
+
+- -?|-h|--help
+
+Prints out a description of how to use the command.
+
+#### Examples
+
+- List all the NuGet configuration file that will be applied, when invoking NuGet command in the current directory.
+
+```
+dotnet nuget config list
+
+C:\Test\Repos\Solution\NuGet.Config
+C:\Test\Repos\NuGet.Config
+C:\Test\NuGet.Config
+C:\Users\username\AppData\Roaming\NuGet\NuGet.Config
+C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config
+C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config
+```
+
+- List all the NuGet configuration file that will be applied, when invoking NuGet command in the specific directory.
+
+```
+dotnet nuget config list  C:\Test\Repos
+
+C:\Test\Repos\NuGet.Config
+C:\Test\NuGet.Config
+C:\Users\username\AppData\Roaming\NuGet\NuGet.Config
+C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config
+C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config
+```
+
+## Future Work
+The `dotnet nuget config list` is a community ask. We will consider adding more commands, like add/update/delete, in the future.
+
+## Open Questions
+1. Do we need to add this into NuGet.exe CLI?
+
+## Considerations
+1. NuGet.exe [config command](https://learn.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-config) is implemented. But there is no `list` command. And the behavior is confusing (the `set` command will set the property which appears last when loading, so sometimes it's not updating the closest NuGet configuration file). Do we want to implement those subcommand(e.g.`set`) in the future in dotnet.exe differently?
```

Comment:
I would say yes to this question. Let's create an issue for that.

Reply:
> And the behavior is confusing (the `set` command will set the property which appears last when loading, so sometimes it's not updating the closest NuGet configuration file). Do we want to implement those subcommand(e.g.`set`) in the future in dotnet.exe differently?

I think that's a discussion for a future point. 

Personally, I'd prefer we guide users towards choosing the right config file instead of NuGet making assumptions. 

I'd be hesitant to have VS and NuGet.exe work differently from dotnet.exe.

Reply:
Thanks for reviewing! Moved it to "Future Work" section as we won't address it in the implementation.